pkg/controller/ipam/antrea_ipam_controller.go

```diff
@@ -52,13 +54,13 @@ const (
 
 	minRetryDelay = 5 * time.Second
 	maxRetryDelay = 300 * time.Second
+
+	garbageCollectionInterval = 10 * time.Minute
 )
 
-// AntreaIPAMController is responsible for IP address cleanup
-// for StatefulSet objects.
-// In future, it will also be responsible for pre-allocation of
-// continuous IP range for StatefulSets that do not have dedicated
-// IP Pool annotation.
+// AntreaIPAMController is responsible for:
+// * reserving continuous IP address space for StatefulSet (if available)
+// * periodical cleanup of IP Pools in case stale addresses are present
```

Comment:
It should work if we do cleanup at Pod deletion and GC at Controller start. But I assume that requires more code and has higher cost (to watch Pods and looks up allocation)? As GC is only for uncommon case, I feel periodic GC is good too.

Reply:
By uncommon case, you mean antrea-agent would clean up the allocation when handling CNI del and the periodical cleanup is only in case that handling fails? If yes, I agree the current approach is good enough.

Reply:
Yes, only when agent stops when CNI del happens.